File: app/src/main/java/com/bigscreen/mangindo/base/MangindoApplication.java
package com.bigscreen.mangindo.base;

import android.app.Application;
import android.util.Log;

import com.bigscreen.mangindo.deps.AppDeps;
import com.bigscreen.mangindo.deps.AppModule;
import com.bigscreen.mangindo.deps.DaggerAppDeps;

public class MangindoApplication extends Application {

    private static final String TAG = MangindoApplication.class.getSimpleName();

    private AppDeps appDeps;

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d(TAG, "Application created.");
        appDeps = DaggerAppDeps.builder().appModule(new AppModule(this)).build();
    }

    @Override
    public void onTerminate() {
        Log.d(TAG, "Application terminated.");
        super.onTerminate();
    }

    public AppDeps getAppDeps() {
        return appDeps;
    }
}

File: app/src/main/java/com/bigscreen/mangindo/base/MangindoApplication.kt
package com.bigscreen.mangindo.base

import android.app.Application
import androidx.appcompat.app.AppCompatDelegate
import com.bigscreen.mangindo.deps.AppDeps
import com.bigscreen.mangindo.deps.AppModule
import com.bigscreen.mangindo.deps.DaggerAppDeps

class MangindoApplication : Application() {

    private lateinit var appDepsX: AppDeps

    override fun onCreate() {
        super.onCreate()
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
        appDepsX = DaggerAppDeps.builder().appModule(AppModule(this)).build()
    }

    fun getAppDeps(): AppDeps {
        return appDepsX
    }
}
